### PR TITLE
Tectonic benchmarks

### PR DESCRIPTION
**Pull request by djspiewak**
This also adds a `Table.parseJson` function which takes a misleading boolean which pretends to handle precise… but doesn't. Support incoming soon! The preliminary results here are really exciting though. Tectonic is between 10% and 90% faster than jawn + qdata + rvalue, and there's plenty of optimization yet to do.

Brings in #3861

### CODE REVIEW

File: yggdrasil/src/main/scala/quasar/yggdrasil/table/ColumnarTableModule.scala [hunk 1]
```diff
@@ -364,6 +364,46 @@ trait ColumnarTableModule
...
+        implicit ec: ExecutionContext)
+        : M[Table] = {
+
+      import fs2.{Chunk, Stream}
+      import tectonic.AsyncParser
+
+      val parser = AsyncParser(new SlicePlate, AsyncParser.ValueStream)
+
+      val absorbed: Stream[IO, Chunk[Slice]] =
+        bytes.chunks.evalScan(Chunk.empty[Slice]) { (_, bc) =>
```
Comment (rintcius) [inline yggdrasil/src/main/scala/quasar/yggdrasil/table/ColumnarTableModule.scala]: looks redundant to call `evalScan` without using the accumulator (ie the `_` in `(_, bc)`). 
Would calling a simpler method like `evalMap` work here?

File: yggdrasilPerf/src/main/scala/quasar/yggdrasil/perf/TableConstructionBenchmark.scala [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        RObject(List.tabulate(keysPerObject)(i =>
+          ("k" + ((s * keysPerObject + i) % keys)) -> scalar
+        ).toMap)))
+
+  def createAndConsumeTable(data: fs2.Stream[IO, Byte], bh: Blackhole): IO[Unit] = {
+    def parseWithJawn: fs2.Stream[IO, RValue] = {
+      implicit val facade = qdata.json.QDataFacade.qdata[RValue]
+      val parser = jawn.AsyncParser[RValue](jawn.AsyncParser.ValueStream)
+
+      val absorbtion =
```
Comment (rintcius) [inline yggdrasilPerf/src/main/scala/quasar/yggdrasil/perf/TableConstructionBenchmark.scala]: `absorption`

File: yggdrasilPerf/src/main/scala/quasar/yggdrasil/perf/TableConstructionBenchmark.scala [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        data.chunks.scan(fs2.Chunk.empty[RValue]) { (_, chunk) =>
```
Comment (rintcius) [inline yggdrasilPerf/src/main/scala/quasar/yggdrasil/perf/TableConstructionBenchmark.scala]: similar here: can we use `data.chunks.map`?

File: yggdrasil/src/main/scala/quasar/yggdrasil/table/ColumnarTableModule.scala (hunk 1 above)
Comment (djspiewak) [reply yggdrasil/src/main/scala/quasar/yggdrasil/table/ColumnarTableModule.scala]: Uh, yes, `evalMap` would definitely work. I'm not sure why I fixated on scan.

File: yggdrasilPerf/src/main/scala/quasar/yggdrasil/perf/TableConstructionBenchmark.scala (hunk 3 above)
Comment (djspiewak) [reply yggdrasilPerf/src/main/scala/quasar/yggdrasil/perf/TableConstructionBenchmark.scala]: Yep

**Pull request merged**